lib/spack/spack/test/cmd/env.py

```diff
@@ -721,91 +721,125 @@ def test_env_with_include_config_files_same_basename():
     assert environment_specs[1].satisfies("mpileaks@2.2")
 
 
-def test_env_with_included_config_file():
-    test_config = """\
+@pytest.fixture(scope="function")
+def packages_file(tmpdir):
+    """Return the path to the packages configuration file."""
+    raw_yaml = """
+packages:
+  mpileaks:
+    version: [2.2]
+"""
+    filename = tmpdir.ensure("testconfig", "packages.yaml")
+    filename.write(raw_yaml)
+    yield filename
+
+
+def mpileaks_env_config(include_path):
+    """Return the contents of an environment that includes the provided
+    path and lists mpileaks as the sole spec."""
+    return """\
 env:
   include:
-  - ./included-config.yaml
+  - {0}
   specs:
   - mpileaks
-"""
+""".format(
+        include_path
+    )
+
+
+def test_env_with_included_config_file(packages_file):
+    """Test inclusion of a relative packages configuration file added to an
+    existing environment."""
+    include_filename = "included-config.yaml"
+    test_config = mpileaks_env_config(os.path.join(".", include_filename))
+
     _env_create("test", StringIO(test_config))
     e = ev.read("test")
 
-    with open(os.path.join(e.path, "included-config.yaml"), "w") as f:
-        f.write(
-            """\
-packages:
-  mpileaks:
-    version: [2.2]
-"""
-        )
+    included_path = os.path.join(e.path, include_filename)
+    fs.rename(packages_file.strpath, included_path)
 
     with e:
         e.concretize()
 
     assert any(x.satisfies("mpileaks@2.2") for x in e._get_environment_specs())
 
 
-def test_env_with_included_config_scope():
+def test_env_with_included_config_file_url(tmpdir, mutable_empty_config, packages_file):
+    """Test configuration inclusion of a file whose path is a URL before
+    the environment is concretized."""
+
+    spack_yaml = tmpdir.join("spack.yaml")
+    with spack_yaml.open("w") as f:
+        f.write("spack:\n  include:\n    - file://{0}\n".format(packages_file))
+
+    env = ev.Environment(tmpdir.strpath)
+    ev.activate(env)
+    scopes = env.included_config_scopes()
+    assert len(scopes) == 1
+
+    cfg = spack.config.get("packages")
+    assert cfg["mpileaks"]["version"] == [2.2]
+
+
+def test_env_with_included_config_missing_file(tmpdir, mutable_empty_config):
+    """Test inclusion of a missing configuration file raises FetchError
+    noting missing file."""
+
+    spack_yaml = tmpdir.join("spack.yaml")
+    missing_file = tmpdir.join("packages.yaml")
+    with spack_yaml.open("w") as f:
+        f.write("spack:\n  include:\n    - {0}\n".format(missing_file.strpath))
+
+    env = ev.Environment(tmpdir.strpath)
+    with pytest.raises(FetchError, match="No such file or directory"):
+        ev.activate(env)
+
+
+def test_env_with_included_config_scope(tmpdir, packages_file):
+    """Test inclusion of a 'remote' package configuration file 'staged' in the
+    environment's configuration stage directory."""
```

Comment:
I'm having trouble relating this docstring to this test:

* I don't see how this is remote: it's a file (and in particular it's not a URL)
* I assume the file might become staged as part of the process but not sure if that's an essential detail (however the environment wants to manage this "under the hood" is fine, not something that needs to be explained here).
* That being said, when I read through the code in `environment.py`, I don't see local files that exist being placed in the config stage directory (that only appears to happen for URLs).

Reply:
I was trying to convey that this test is intended to represent the case of including a previously staged (from a remote URL) package configuration file.  Will re-word.

Local files are not staged since they are already available on the file system so their existing file path is sufficient.

Reply:
Modified the comment.